homeassistant/components/zimi/coordinator.py

```diff
@@ -0,0 +1,96 @@
+"""Zimi Controller coordinator class device."""
+
+import logging
+import pprint
+
+from zcc import ControlPoint, ControlPointDescription, ControlPointError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_TIMEOUT, CONF_VERBOSITY, CONF_WATCHDOG, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZimiCoordinator(DataUpdateCoordinator):
+    """Coordinates a single Zimi Controller hub.
+
+    Initial list of devices is fetched in _async_setup() but all subsequent data
+    is pushed from the zcc helper and handled in the Entities.
+    """
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=entry,
+            name="Zimi Cloud Connect",
+            always_update=False,
+        )
+
+        if self.config_entry:
+            _LOGGER.debug("Initialising:\n%s", pprint.pformat(self.config_entry.data))
+
+        self.api: ControlPoint | None = None
+
+    async def _async_setup(self):
+        """Set up the coordinator."""
+
+        try:
+            _LOGGER.info(
```

Comment:
apologies, I wasn't aware about the subscription logic. Then you should just be able to cut, paste this code from here to __init__.py, `async_setup_entry` and remove the coordonator

Reply:
On second thought, leave the coordinator here until we're set about your updates

Reply:
I think we need to keep coordinator.   It is possible to have multiple ZCC devices on the network, each with their own set of connected devices.   I would like to have a coordinator instance for each one.